tests/chainer_tests/training_tests/triggers_tests/test_interval_trigger.py

```diff
@@ -30,51 +33,81 @@ def epoch_detail(self):
-
-
-class TestEpochIntervalTrigger(unittest.TestCase):
-
-    def test_epoch_interval_trigger(self):
-        updater = DummyUpdater(iters_per_epoch=5)
-        trigger = training.trigger.IntervalTrigger(1, 'epoch')
-        expected = [False, False, False, False, True, False, False]
-        _test_trigger(self, updater, trigger, expected)
-
-
-class TestFractionalEpochIntervalTrigger(unittest.TestCase):
-
-    def test_epoch_interval_trigger(self):
-        updater = DummyUpdater(iters_per_epoch=2)
-        trigger = training.trigger.IntervalTrigger(1.5, 'epoch')
-        expected = [False, False, True, False, False, True, False]
-        _test_trigger(self, updater, trigger, expected)
-
-
-class TestUnalignedEpochIntervalTrigger(unittest.TestCase):
-
-    def test_unaligned_epoch_interval_trigger(self):
-        updater = DummyUpdater(iters_per_epoch=2.5)
-        trigger = training.trigger.IntervalTrigger(1, 'epoch')
-        expected = [False, False, True, False, True, False, False]
-        _test_trigger(self, updater, trigger, expected)
+    def serialize(self, serializer):
+        self.iteration = serializer('iteration', self.iteration)
+
+
+@testing.parameterize(*testing.product_dict([
+    # iteration
+    {
+        'iters_per_epoch': 5, 'interval': (2, 'iteration'), 'resume': 4,
+        'expected': [False, True, False, True, False, True, False]},
+    # basic epoch
+    {
+        'iters_per_epoch': 1, 'interval': (3, 'epoch'), 'resume': 4,
+        'expected': [False, False, True, False, False, True, False]},
+    # fractional epoch
+    {
+        'iters_per_epoch': 2, 'interval': (1.5, 'epoch'), 'resume': 4,
+        'expected': [False, False, True, False, False, True, False]},
+    # unaligned epoch
+    {
+        'iters_per_epoch': 2.5, 'interval': (1, 'epoch'), 'resume': 3,
+        'expected': [False, False, True, False, True, False, False]},
+    # tiny epoch
+    {
+        'iters_per_epoch': 0.5, 'interval': (1, 'epoch'), 'resume': 4,
+        'expected': [True, True, True, True, True, True, True]},
+]))
+class TestTrigger(unittest.TestCase):
+
+    def test_trigger(self):
+        trigger = training.trigger.IntervalTrigger(*self.interval)
+        updater = DummyUpdater(self.iters_per_epoch)
+        trainer = training.Trainer(updater)
+
+        for expected in self.expected:
+            updater.update()
+            self.assertEqual(trigger(trainer), expected)
+
+    def test_stop_trigger(self):
+        trigger = training.trigger.IntervalTrigger(*self.interval)
+        updater = DummyUpdater(self.iters_per_epoch)
+        trainer = training.Trainer(updater)
+        for expected in [False] + self.expected:
+            self.assertEqual(trigger(trainer), expected)
+            if expected:
+                break
+            updater.update()
+
+    def test_resumed_trigger(self):
+        temp_dir = tempfile.mkdtemp()
+        temp_file = os.path.join(temp_dir, 'temp.npz')
```

Comment:
I think [`NamedTemporaryFile`](http://docs.python.jp/2/library/tempfile.html#tempfile.NamedTemporaryFile) is simpler, if it is allowed that the temporary file does not have `npz` suffix, because you do not have to create a temporary directory and the temporary file is automatically removed.